packages/jest-cli/src/cli/index.js

```diff
@@ -103,12 +104,41 @@ export const runCLI = async (
   const {openHandles} = results;
 
   if (openHandles && openHandles.length) {
-    const openHandlesString = pluralize('open handle', openHandles.length, 's');
+    const formatted = formatHandleErrors(openHandles, configs[0]);
+
+    const stacks = new Set();
+
+    // E.g. timeouts might give multiple traces to the same line of code
+    // This hairy filtering tries to remove entries with duplicate stack traces
+    const uniqueErrors = formatted.filter(handle => {
```

Comment:
should we exatract this into a separate module?
i a little worried about introducing so much complex logic right in `index.js` :P

Reply:
yeah, I can extract it

Reply:
done now